From f5a7a6053f2dad7e50e222ad5666abcd3f49ee6e Mon Sep 17 00:00:00 2001
Message-Id: <f5a7a6053f2dad7e50e222ad5666abcd3f49ee6e.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: arsharma <ankitprasad.r.sharma@intel.com>
Date: Thu, 4 Sep 2014 15:14:15 +0530
Subject: [PATCH 268/312] MUST_REBASE [VPG]: drm/i915: Fix for DPST memleak
 issue

This patch fixes memleak issue from DPST where drm_display_mode structure
pointer was not being freed before returning from i915_dpst_get_resolution,
so freeing drm_display_mode structure pointer before returning

Issue: GMIN-1877
Change-Id: If3d13401f976679c98c2e82d1da878dbb8a777fd
Signed-off-by: arsharma <ankitprasad.r.sharma@intel.com>
---
 drivers/gpu/drm/i915/i915_dpst.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dpst.c b/drivers/gpu/drm/i915/i915_dpst.c
index c4478de..5979f88 100644
--- a/drivers/gpu/drm/i915/i915_dpst.c
+++ b/drivers/gpu/drm/i915/i915_dpst.c
@@ -353,6 +353,7 @@ i915_dpst_get_resolution(struct drm_device *dev)
 	struct drm_i915_private *dev_priv = dev->dev_private;
 	struct drm_crtc *crtc;
 	struct drm_display_mode *mode;
+	u32 resolution = 0;
 
 	/* Get information about current display mode */
 	crtc = intel_get_crtc_for_pipe(dev, dev_priv->dpst.pipe);
@@ -361,9 +362,10 @@ i915_dpst_get_resolution(struct drm_device *dev)
 
 	mode = intel_crtc_mode_get(dev, crtc);
 	if (mode)
-		return  mode->hdisplay * mode->vdisplay;
+		resolution = mode->hdisplay * mode->vdisplay;
 
-	return 0;
+	kfree(mode);
+	return resolution;
 }
 
 static int
-- 
1.7.9.5

